core/node/node_framework/src/implementations/layers/web3_api/tx_sink/master_pool_sink.rs

```diff
@@ -35,8 +43,18 @@ impl WiringLayer for MasterPoolSinkLayer {
 
     async fn wire(self, input: Self::Input) -> Result<Self::Output, WiringError> {
         let pool = input.master_pool.get().await?;
-        Ok(Output {
-            tx_sink: MasterPoolSink::new(pool).into(),
-        })
+        let master_pool_sink = MasterPoolSink::new(pool.clone());
+
+        let tx_sink = if self.rpc_config.deployment_allowlist_sink {
+            let allowlist_service = AllowListService::new(
+                self.rpc_config.http_file_url.unwrap_or_default(),
```

Comment:
`unwrap_or_default` doesn't look reasonable here and below; "" is not a valid URL, and `Duration::ZERO` is not a reasonable refresh interval.

- `http_file_url` can be used to signal whether allowlisting is enabled (if it's `Some(_)`, it is, if it's `None`, it isn't).
- `refresh_interval_secs` field should be private and have a public getter, as [some other config fields](https://github.com/matter-labs/zksync-era/blob/ed4926f4cd0705ad6f1ed57ff5ee2c8c79af4987/core/lib/config/src/configs/api.rs#L290).